packages/sdk-core/src/modules/data/BacktraceDataBuilder.ts

```diff
@@ -68,4 +51,32 @@ export class BacktraceDataBuilder {
 
         return result;
     }
+
+    private getThreads(report: BacktraceReport) {
+        const threads: Record<string, BacktraceStackTrace> = {};
+        let detectedDebugIdentifier = false;
+
+        for (const [name, traceInfo] of Object.entries(report.stackTrace)) {
+            const { message, stack } = traceInfo;
+            const stackFrames = this._stackTraceConverter.convert(stack, message);
+            threads[name] = {
+                fault: name === this.MAIN_THREAD_NAME,
+                name,
+                stack: stackFrames,
+            };
+
+            for (const frame of stackFrames) {
+                const debugIdentifier = this._debugIdProvider.getDebugId(frame.library);
+                if (!debugIdentifier) {
+                    continue;
+                }
+                detectedDebugIdentifier = true;
+                frame.debug_identifier = debugIdentifier;
+            }
```

Comment:
it feels more natural to set the thread once we add debug_identifiers. The current approach should still work, however it's easy to mess up something when we modify the code and we don't have the same reference to the object. Can we set the thread after `for (const frame of stackFrames) {` loop?

Reply:
Fair point, fixed